### PR TITLE
[RECO] [CLANG] Fixed unused-but-set-variable warnings

### PR DESCRIPTION
**Pull request by smuzaffar**
This PR fixes `unused-but-set-variable` warnings which we get with LLVM14 in CLANG IBS

### CODE REVIEW

**Comment by tvami** [comment]
@cmsbuild , please test

File: RecoLocalTracker/SiPixelClusterizer/test/TestClusters.cc [hunk 1]
```diff
@@ -2304,7 +2304,7 @@ void TestClusters::analyze(const edm::Event &e, const edm::EventSetup &es) {
   int numOfPixPerDisk3 = 0;
   int numOfPixPerDisk4 = 0;
 
-  float avCharge1 = 0., avCharge2 = 0., avCharge3 = 0., avCharge4 = 0., avCharge5 = 0.;
+  //float avCharge1 = 0., avCharge2 = 0., avCharge3 = 0., avCharge4 = 0., avCharge5 = 0.;
```
Comment (clacaputo) [inline RecoLocalTracker/SiPixelClusterizer/test/TestClusters.cc:2307]: Wouldn't be better to remove it?
(same comment for all the others commented lines)

Comment (smuzaffar) [reply RecoLocalTracker/SiPixelClusterizer/test/TestClusters.cc:2307]: @clacaputo , I did not delete it as there is some commented out code ( e.g. see code after https://github.com/cms-sw/cmssw/blob/19451fc97b6dda130b29624d4540575f1e43f250/RecoLocalTracker/SiPixelClusterizer/test/TestClusters.cc#L3536 ) which uses these variables.

**Comment by perrotta** [comment]
+1

**Pull request merged**